src/datadog/tracer_config.cpp

```diff
@@ -742,38 +848,45 @@ Expected<FinalizedTracerConfig> finalize_config(
   TraceSamplerConfig ts;
   if (env->trace_sampler.sample_rate) {
     ts.sample_rate = env->trace_sampler.sample_rate;
+    origin = ConfigMetadata::Origin::ENVIRONMENT_VARIABLE;
   } else if (user_config.trace_sampler.sample_rate) {
     ts.sample_rate = user_config.trace_sampler.sample_rate;
+    origin = ConfigMetadata::Origin::CODE;
   }
 
+  final_cfg.metadata[ConfigName::TRACE_SAMPLING_RATE] =
+      ConfigMetadata(ConfigName::TRACE_SAMPLING_RATE,
+                     to_string(ts.sample_rate.value_or(1.0)), origin);
+
+  // TS: Rules
```

Comment:
I'd prefer a longer variable name than `ts`. But, if you are going with `ts`, then at least use the full name in comments.
```c++
// trace sampler rules
...
// trace sampler max per second
```